src/controllers/portfolio/portfolio.ts

```diff
@@ -692,6 +700,34 @@ export class PortfolioController extends EventEmitter {
     this.emitUpdate()
   }
 
+  addTokensToBeLearned(tokenAddresses: string[], networkId: NetworkId) {
+    if (!tokenAddresses) return false
+    if (!this.#toBeLearnedTokens) this.#toBeLearnedTokens = {}
+    if (!this.#toBeLearnedTokens[networkId]) this.#toBeLearnedTokens[networkId] = []
+
+    let networkToBeLearnedTokens = this.#toBeLearnedTokens[networkId] || []
+    const networkLearnedTokens = this.#previousHints.learnedTokens[networkId] || {}
+
+    const alreadyLearned = networkToBeLearnedTokens.map((addr) => getAddress(addr))
+    const alreadyLearnedTokens = Object.keys(networkLearnedTokens).map((addr) => getAddress(addr))
+
+    const tokensToLearn = tokenAddresses.filter((address) => {
+      const normalizedAddress = getAddress(address)
+      return (
+        !alreadyLearned.includes(normalizedAddress) &&
+        !alreadyLearnedTokens.includes(normalizedAddress) &&
+        !address === ZeroAddress
+      )
+    })
+
+    if (!tokensToLearn.length) return false
+
+    networkToBeLearnedTokens = [...tokensToLearn, ...networkToBeLearnedTokens]
+
+    this.#toBeLearnedTokens[networkId] = networkToBeLearnedTokens
```

Comment:
Imo this logic is mostly unnecessary. I don't see a problem in adding all passed `tokenAddresses` to `toBeLearnedTokens`. After all, hints are deduped in portfolio lib and learnTokens doesn't allow duplicates.